server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RSpamDScanner.java

```diff
@@ -0,0 +1,116 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.rspamd.client.RSpamDClientConfiguration;
+import org.apache.james.rspamd.client.RSpamDHttpClient;
+import org.apache.james.rspamd.model.AnalysisResult;
+import org.apache.mailet.Attribute;
+import org.apache.mailet.AttributeName;
+import org.apache.mailet.AttributeValue;
+import org.apache.mailet.Mail;
+import org.apache.mailet.PerRecipientHeaders;
+import org.apache.mailet.base.GenericMailet;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+public class RSpamDScanner extends GenericMailet {
+    public static final AttributeName FLAG_MAIL = AttributeName.of("org.apache.james.spamassassin.flag");
```

Comment:
Should we still keep using this attribute?
Or another for less confuse 
eg: `org.apache.james.rspamd.flag`
But if we use another attribute, we need to modify the `IsMarkedAsSpam` matcher

Reply:
I would agree on something more generic if we have more than one spam implem yes
like `org.apache.james.spam.flag` maybe?

Reply:
I thought again, 
The attribute should be unique, eg case we use multi spam 3rd (both rspamd and spamassian)
IMO, we will keep `org.apache.james.spamassassin.x` for spamassian,
`org.apache.james.rspamd.x` for rspamd. 
And modify the `IsMarkedAsSpam` for reading attributeName from mailet XML configuration file.

Reply:
Ok to have distinct flags
However let's be honest, there is nothing logical for a environment to use 2 anti spam third party softwares at the same time :)